<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PFWidget</class>
 <widget class="QWidget" name="PFWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>913</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="1,0">
   <item>
    <widget class="QWidget" name="playerWidget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,0,0">
      <item>
       <widget class="QLabel" name="display_label">
        <property name="styleSheet">
         <string notr="true">Background-color: #000;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSlider" name="horizontalSlider">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <spacer name="buttonsLeft_horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="browse_pushButton">
          <property name="font">
           <font>
            <family>Futura</family>
            <pointsize>14</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Browse</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="start_pushButton">
          <property name="font">
           <font>
            <family>Futura</family>
            <pointsize>14</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Start</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="pause_pushButton">
          <property name="font">
           <font>
            <family>Futura</family>
            <pointsize>14</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Pause</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="saveTarget_pushButton">
          <property name="font">
           <font>
            <family>Futura</family>
            <pointsize>14</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="text">
           <string>Save Target</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="buttonsRight_horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QGridLayout" name="gridLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item row="0" column="0">
       <widget class="QGroupBox" name="groupBox">
        <property name="title">
         <string>PF Settings</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <widget class="QLabel" name="particleCountLabel">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string># Particles</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="particleCountSpinBox">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>100</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="iterCountLabel">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string># Iterations</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="iterCountSpinBox">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>10</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="particleWidthLbel">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Particle Width</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="particleWidthLSpinBox">
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>55</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="HistogramSizelabel">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Histogram Size</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="histogramSizespinBox">
           <property name="minimum">
            <number>8</number>
           </property>
           <property name="maximum">
            <number>256</number>
           </property>
           <property name="singleStep">
            <number>4</number>
           </property>
           <property name="value">
            <number>64</number>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Top N particles</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QSlider" name="particlesToDisplaySlider">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="particlesToDisplayLabel">
             <property name="minimumSize">
              <size>
               <width>30</width>
               <height>0</height>
              </size>
             </property>
             <property name="text">
              <string>0</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_4">
           <item>
            <widget class="QLabel" name="label_2">
             <property name="font">
              <font>
               <family>Futura</family>
               <pointsize>14</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Ratio: </string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="top_n_ratio_label">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="label_3">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Targets :</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="targets_ListWidget">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="maximumSize">
            <size>
             <width>150</width>
             <height>10000</height>
            </size>
           </property>
           <property name="font">
            <font>
             <family>Helvetica Neue</family>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="setupPF_pushButton">
           <property name="font">
            <font>
             <family>Futura</family>
             <pointsize>14</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Setup Particle Filter</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>browse_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>PFWidget</receiver>
   <slot>onActionOpen()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>604</y>
    </hint>
    <hint type="destinationlabel">
     <x>163</x>
     <y>710</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>start_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>PFWidget</receiver>
   <slot>onActionPlay()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>338</x>
     <y>602</y>
    </hint>
    <hint type="destinationlabel">
     <x>353</x>
     <y>663</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pause_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>PFWidget</receiver>
   <slot>onActionPause()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>434</x>
     <y>603</y>
    </hint>
    <hint type="destinationlabel">
     <x>450</x>
     <y>683</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>sliderPressed()</signal>
   <receiver>PFWidget</receiver>
   <slot>onHorizontalSliderPressed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>176</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>193</x>
     <y>724</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>sliderReleased()</signal>
   <receiver>PFWidget</receiver>
   <slot>onHorizontalSliderReleased()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>48</x>
     <y>727</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particleCountSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onParticleCountChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>1140</x>
     <y>77</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>iterCountSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onIterationCountChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>148</y>
    </hint>
    <hint type="destinationlabel">
     <x>1004</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particleWidthLSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onParticleWidthChange(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>882</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>1091</x>
     <y>185</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>particlesToDisplaySlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onSetParticleCountSliderMoved(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>845</x>
     <y>313</y>
    </hint>
    <hint type="destinationlabel">
     <x>1065</x>
     <y>233</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>saveTarget_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>PFWidget</receiver>
   <slot>onActionSaveTarget()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>506</x>
     <y>613</y>
    </hint>
    <hint type="destinationlabel">
     <x>507</x>
     <y>699</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>histogramSizespinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onHistSizeChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>821</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>930</x>
     <y>480</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>setupPF_pushButton</sender>
   <signal>clicked()</signal>
   <receiver>PFWidget</receiver>
   <slot>onActionSetupPF()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>759</x>
     <y>598</y>
    </hint>
    <hint type="destinationlabel">
     <x>758</x>
     <y>673</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>horizontalSlider</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>PFWidget</receiver>
   <slot>onHorizontalSliderMoved(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>106</x>
     <y>578</y>
    </hint>
    <hint type="destinationlabel">
     <x>146</x>
     <y>769</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onActionOpen()</slot>
  <slot>onActionPlay()</slot>
  <slot>onActionPause()</slot>
  <slot>onHorizontalSliderPressed()</slot>
  <slot>onHorizontalSliderReleased()</slot>
  <slot>onHorizontalSliderMoved(int)</slot>
  <slot>onParticleCountChange(int)</slot>
  <slot>onIterationCountChange(int)</slot>
  <slot>onParticleWidthChange(int)</slot>
  <slot>onSetParticleCountSliderMoved(int)</slot>
  <slot>onActionSaveTarget()</slot>
  <slot>onHistSizeChanged(int)</slot>
  <slot>onActionSetupPF()</slot>
 </slots>
</ui>
